# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/workers/remoteworkers/PRemoteWorkerDebugger.ipdl
# Commit: 01fe98c08d10
# Full Hash: 01fe98c08d102c28e3a61209ea31834eff83202a
# Author: Eden Chuang <echuang@mozilla.com>
# Date: 2025-03-25 21:02:33
# Regressor Bug: 1899503
# File Overlap Count: 4
# Description:
#   Bug 1899503 - P3 PRemoteWorkerDebugger binding implementation. r=asuth
#   
#   This patch implements the PRemoteWorkerDebugger binding.
#   
#   We are trying to keep the same logic debugger registration/unregistration to minimize the difference between the local debugger(the debugger on the content process main thread)
# ==============================================================================

diff -r 9bd2a82481f4 -r 01fe98c08d10 dom/workers/remoteworkers/PRemoteWorkerDebugger.ipdl
--- a/dom/workers/remoteworkers/PRemoteWorkerDebugger.ipdl	Tue Mar 25 09:13:20 2025 +0000
+++ b/dom/workers/remoteworkers/PRemoteWorkerDebugger.ipdl	Tue Mar 25 09:13:20 2025 +0000
@@ -24,6 +24,9 @@
   async RemoveWindowID(uint64_t aWindowID);
 
 child:
+  async RegisterDone();
+  async UnregisterDone();
+
   async Initialize(nsString aURL);
   async PostMessage(nsString aMessage);
   async SetDebuggerReady(bool aReady);